example/histogram.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2023 Jiri Nytra (jiri.nytra at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux.hpp>
+
+#include <cstddef>
+#include <ostream>
+#include <string>
+#include <random>
+#include <iostream>
+#include <iomanip>
+#include <map>
+
+template<typename T>
+flux::generator<int> randu(T min, T max)
+{
+    std::random_device rng;
+    std::uniform_int_distribution dist(min, max);
+
+    while (true)
+        co_yield dist(rng);
+}
+
+template<typename T>
+flux::generator<int> randn(T mean, T stddev = 1.0)
+{
+    std::random_device rng;
+    std::normal_distribution dist(mean, stddev);
+
+    while (true)
+        co_yield std::round(dist(rng));
+}
+
+using hist_t = std::map<int, int>;
+
+auto to_histogram = [](auto&& so_far, auto x)
+{
+    ++so_far[std::round(x)];
+    return FLUX_FWD(so_far);
+};
+
+std::ostream& operator<<(std::ostream& s, const hist_t& hist)
+{
+    for (auto [bin, count]: hist)
+        s << std::setw(2) << bin << ' ' << std::string(count / 200, '*') << '\n';
+    return s;
+};
```

Comment:
This makes sense. Braces around single statements are not a problem. Are there other rules for formatting source code? Maybe we can add a `.clang-format' file with formatting rules.

Reply:
I think you already follow most of the rules! I try not to go too far over 80 columns but it's not a strict rule if going a little over makes the code more readable.

I've tried clang format but I found it didn't understand concepts very well last time I tried, and ended up making the source code look a lot worse -- but perhaps newer versions are better?

Reply:
I am also not a fan of automatic formatting. Especially in ranges/sequences it can be misleading. I will try to follow the rules in the current code as much as possible. Feel free to let me know if anything is wrong.

Reply:
You've been doing a great job so far!